packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -211,6 +211,15 @@ class HotRunner extends ResidentRunner {
       }
     }
 
+    if (debuggingOptions.fastStart) {
```

Comment:
Should this check that the device supports fastStart if we haven't thrown a tool exit?

Reply:
Addressed above